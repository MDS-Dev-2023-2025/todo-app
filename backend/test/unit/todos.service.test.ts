import { beforeEach, describe, it, expect } from "@jest/globals";
import { TodoService } from "../../src/services/todo.service";

describe("TodoService", () => {
  let service: TodoService;

  beforeEach(() => {
    service = new TodoService();
  });

  it("should create a todo", async () => {
    const todo = await service.createTodo({ title: "Test Todo" });
    expect(todo).toMatchSnapshot();
  });

  it("should return all todos", async () => {
    await service.createTodo({ title: "Todo 1" });
    const todos = await service.getAllTodos();
    expect(todos.length).toBe(1);
  });

  it("should update a todo", async () => {
    const todo = await service.createTodo({ title: "Initial" });
    const updated = await service.updateTodo(todo.id, { completed: true });
    expect(updated.completed).toBe(true);
  });

  it("should delete a todo", async () => {
    const todo = await service.createTodo({ title: "To Delete" });
    await service.deleteTodo(todo.id);
    const todos = await service.getAllTodos();
    expect(todos).toHaveLength(0);
  });

  it("should throw an error if the todo to delete is not found", async () => {
    await expect(service.deleteTodo("nonexistent")).rejects.toThrow(
      "Todo not found"
    );
  });
});
