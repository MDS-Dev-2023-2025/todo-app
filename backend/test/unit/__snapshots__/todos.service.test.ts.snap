// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoService should create a todo 1`] = `
{
  "completed": false,
  "id": "1749110306824",
  "title": "Test Todo",
}
`;
